extensions/json-jackson/src/main/java/io/deephaven/json/jackson/Parsing.java

```diff
@@ -0,0 +1,303 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.json.jackson;
+
+import ch.randelshofer.fastdoubleparser.JavaBigDecimalParser;
+import ch.randelshofer.fastdoubleparser.JavaBigIntegerParser;
+import ch.randelshofer.fastdoubleparser.JavaDoubleParser;
+import ch.randelshofer.fastdoubleparser.JavaFloatParser;
+import com.fasterxml.jackson.core.JsonLocation;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.StreamReadFeature;
+import io.deephaven.util.BooleanUtils;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.CharBuffer;
+
+final class Parsing {
+
+    static void assertNoCurrentToken(JsonParser parser) {
+        if (parser.hasCurrentToken()) {
+            throw new IllegalStateException(
+                    String.format("Expected no current token. actual=%s", parser.currentToken()));
+        }
+    }
+
+    static void assertNextToken(JsonParser parser, JsonToken expected) throws IOException {
+        final JsonToken actual = parser.nextToken();
+        if (actual != expected) {
+            throw new IllegalStateException(
+                    String.format("Unexpected next token. expected=%s, actual=%s", expected, actual));
+        }
+    }
+
+    static void assertCurrentToken(JsonParser parser, JsonToken expected) {
+        if (!parser.hasToken(expected)) {
+            throw new IllegalStateException(
+                    String.format("Unexpected current token. expected=%s, actual=%s", expected, parser.currentToken()));
+        }
+    }
+
+    static CharSequence textAsCharSequence(JsonParser parser) throws IOException {
+        return parser.hasTextCharacters()
+                ? CharBuffer.wrap(parser.getTextCharacters(), parser.getTextOffset(), parser.getTextLength())
+                : parser.getText();
+    }
+
+    static class UnexpectedToken extends JsonProcessingException {
```

Comment:
I don't love nest exceptions anymore. They make code hard to read. Also, why aren't we throwing this from the assert methods?